google/cloud/firestore_v1/base_query.py

```diff
@@ -244,6 +246,39 @@ def select(self, field_paths: Iterable[str]) -> "BaseQuery":
             all_descendants=self._all_descendants,
         )
 
+    def _copy(
+        self,
+        *,
+        projection: Optional[query.StructuredQuery.Projection] = _not_passed,
+        field_filters: Optional[Tuple[query.StructuredQuery.FieldFilter]] = _not_passed,
+        orders: Optional[Tuple[query.StructuredQuery.Order]] = _not_passed,
+        limit: Optional[int] = _not_passed,
+        limit_to_last: Optional[bool] = _not_passed,
+        offset: Optional[int] = _not_passed,
+        start_at: Optional[Tuple[dict, bool]] = _not_passed,
+        end_at: Optional[Tuple[dict, bool]] = _not_passed,
+        all_descendants: Optional[bool] = _not_passed,
+    ) -> "BaseQuery":
+        return self.__class__(
+            self._parent,
+            projection=self._evaluate_param(projection, self._projection),
+            field_filters=self._evaluate_param(field_filters, self._field_filters),
+            orders=self._evaluate_param(orders, self._orders),
+            limit=self._evaluate_param(limit, self._limit),
+            limit_to_last=self._evaluate_param(limit_to_last, self._limit_to_last),
+            offset=self._evaluate_param(offset, self._offset),
+            start_at=self._evaluate_param(start_at, self._start_at),
+            end_at=self._evaluate_param(end_at, self._end_at),
+            all_descendants=self._evaluate_param(
+                all_descendants, self._all_descendants
+            ),
+        )
+
+    def _evaluate_param(self, value, fallback_value):
+        """Helper which allows `None` to be passed into `copy` and be set on the
+        copy instead of being misinterpreted as an unpassed parameter."""
+        return value if value != _not_passed else fallback_value
```

Comment:
Oops, sorry, should be `value is not _not_passed`. (You're testing that it is that specific object, not an equivalent object.)

Reply:
I'm not certain where this would materially change outcomes, since `object() != object()`, but, I've changed it over to the `id()`-based check of `is not` to be safe.

Reply:
It's more about saying what you mean, than the code working or not. A linter, for example, would give a warning if you said `value != None` and this is the same scenario.